File: scripts/calculate-logo-proportions.js
// Orson Vision Logo Proportion Calculator
// Based on original logo: 600x200px

const ORIGINAL_LOGO = {
  width: 600,
  height: 200,
  // Left bracket positioning
  leftBracket: {
    vertical: { x: 120, y: 60, width: 10, height: 100 },
    topHorizontal: { x: 120, y: 60, width: 50, height: 10 },
    bottomHorizontal: { x: 120, y: 150, width: 50, height: 10 },
  },
  // Right bracket positioning
  rightBracket: {
    vertical: { x: 470, y: 60, width: 10, height: 100 },
    topHorizontal: { x: 430, y: 60, width: 50, height: 10 },
    bottomHorizontal: { x: 430, y: 150, width: 50, height: 10 },
  },
  // Text positioning
  text: {
    orson: { x: 300, y: 120, fontSize: 32, fontWeight: 600 },
    vision: { x: 300, y: 150, fontSize: 24, fontWeight: 400 },
    tagline: { x: 300, y: 180, fontSize: 14, fontWeight: 300 },
  },
};

function calculateProportions(targetWidth, targetHeight) {
  const scaleX = targetWidth / ORIGINAL_LOGO.width;
  const scaleY = targetHeight / ORIGINAL_LOGO.height;

  // Use the smaller scale to maintain aspect ratio
  const scale = Math.min(scaleX, scaleY);

  console.log(`\n=== Logo Proportion Calculator ===`);
  console.log(
    `Original logo: ${ORIGINAL_LOGO.width}x${ORIGINAL_LOGO.height}px`
  );
  console.log(`Target size: ${targetWidth}x${targetHeight}px`);
  console.log(`Scale factor: ${scale.toFixed(3)}`);

  const scaled = {
    width: targetWidth,
    height: targetHeight,
    scale: scale,
    leftBracket: {
      vertical: {
        x: Math.round(ORIGINAL_LOGO.leftBracket.vertical.x * scale),
        y: Math.round(ORIGINAL_LOGO.leftBracket.vertical.y * scale),
        width: Math.round(ORIGINAL_LOGO.leftBracket.vertical.width * scale),
        height: Math.round(ORIGINAL_LOGO.leftBracket.vertical.height * scale),
      },
      topHorizontal: {
        x: Math.round(ORIGINAL_LOGO.leftBracket.topHorizontal.x * scale),
        y: Math.round(ORIGINAL_LOGO.leftBracket.topHorizontal.y * scale),
        width: Math.round(
          ORIGINAL_LOGO.leftBracket.topHorizontal.width * scale
        ),
        height: Math.round(
          ORIGINAL_LOGO.leftBracket.topHorizontal.height * scale
        ),
      },
      bottomHorizontal: {
        x: Math.round(ORIGINAL_LOGO.leftBracket.bottomHorizontal.x * scale),
        y: Math.round(ORIGINAL_LOGO.leftBracket.bottomHorizontal.y * scale),
        width: Math.round(
          ORIGINAL_LOGO.leftBracket.bottomHorizontal.width * scale
        ),
        height: Math.round(
          ORIGINAL_LOGO.leftBracket.bottomHorizontal.height * scale
        ),
      },
    },
    rightBracket: {
      vertical: {
        x: Math.round(ORIGINAL_LOGO.rightBracket.vertical.x * scale),
        y: Math.round(ORIGINAL_LOGO.rightBracket.vertical.y * scale),
        width: Math.round(ORIGINAL_LOGO.rightBracket.vertical.width * scale),
        height: Math.round(ORIGINAL_LOGO.rightBracket.vertical.height * scale),
      },
      topHorizontal: {
        x: Math.round(ORIGINAL_LOGO.rightBracket.topHorizontal.x * scale),
        y: Math.round(ORIGINAL_LOGO.rightBracket.topHorizontal.y * scale),
        width: Math.round(
          ORIGINAL_LOGO.rightBracket.topHorizontal.width * scale
        ),
        height: Math.round(
          ORIGINAL_LOGO.rightBracket.topHorizontal.height * scale
        ),
      },
      bottomHorizontal: {
        x: Math.round(ORIGINAL_LOGO.rightBracket.bottomHorizontal.x * scale),
        y: Math.round(ORIGINAL_LOGO.rightBracket.bottomHorizontal.y * scale),
        width: Math.round(
          ORIGINAL_LOGO.rightBracket.bottomHorizontal.width * scale
        ),
        height: Math.round(
          ORIGINAL_LOGO.rightBracket.bottomHorizontal.height * scale
        ),
      },
    },
    text: {
      orson: {
        x: Math.round(ORIGINAL_LOGO.text.orson.x * scale),
        y: Math.round(ORIGINAL_LOGO.text.orson.y * scale),
        fontSize: Math.round(ORIGINAL_LOGO.text.orson.fontSize * scale),
        fontWeight: ORIGINAL_LOGO.text.orson.fontWeight,
      },
      vision: {
        x: Math.round(ORIGINAL_LOGO.text.vision.x * scale),
        y: Math.round(ORIGINAL_LOGO.text.vision.y * scale),
        fontSize: Math.round(ORIGINAL_LOGO.text.vision.fontSize * scale),
        fontWeight: ORIGINAL_LOGO.text.vision.fontWeight,
      },
      tagline: {
        x: Math.round(ORIGINAL_LOGO.text.tagline.x * scale),
        y: Math.round(ORIGINAL_LOGO.text.tagline.y * scale),
        fontSize: Math.round(ORIGINAL_LOGO.text.tagline.fontSize * scale),
        fontWeight: ORIGINAL_LOGO.text.tagline.fontWeight,
      },
    },
  };

  console.log(`\n=== Scaled Proportions ===`);
  console.log(`Left Bracket:`);
  console.log(
    `  Vertical: x=${scaled.leftBracket.vertical.x}, y=${scaled.leftBracket.vertical.y}, width=${scaled.leftBracket.vertical.width}, height=${scaled.leftBracket.vertical.height}`
  );
  console.log(
    `  Top Horizontal: x=${scaled.leftBracket.topHorizontal.x}, y=${scaled.leftBracket.topHorizontal.y}, width=${scaled.leftBracket.topHorizontal.width}, height=${scaled.leftBracket.topHorizontal.height}`
  );
  console.log(
    `  Bottom Horizontal: x=${scaled.leftBracket.bottomHorizontal.x}, y=${scaled.leftBracket.bottomHorizontal.y}, width=${scaled.leftBracket.bottomHorizontal.width}, height=${scaled.leftBracket.bottomHorizontal.height}`
  );

  console.log(`\nRight Bracket:`);
  console.log(
    `  Vertical: x=${scaled.rightBracket.vertical.x}, y=${scaled.rightBracket.vertical.y}, width=${scaled.rightBracket.vertical.width}, height=${scaled.rightBracket.vertical.height}`
  );
  console.log(
    `  Top Horizontal: x=${scaled.rightBracket.topHorizontal.x}, y=${scaled.rightBracket.topHorizontal.y}, width=${scaled.rightBracket.topHorizontal.width}, height=${scaled.rightBracket.topHorizontal.height}`
  );
  console.log(
    `  Bottom Horizontal: x=${scaled.rightBracket.bottomHorizontal.x}, y=${scaled.rightBracket.bottomHorizontal.y}, width=${scaled.rightBracket.bottomHorizontal.width}, height=${scaled.rightBracket.bottomHorizontal.height}`
  );

  console.log(`\nText:`);
  console.log(
    `  ORSON: x=${scaled.text.orson.x}, y=${scaled.text.orson.y}, fontSize=${scaled.text.orson.fontSize}`
  );
  console.log(
    `  VISION: x=${scaled.text.vision.x}, y=${scaled.text.vision.y}, fontSize=${scaled.text.vision.fontSize}`
  );
  console.log(
    `  Tagline: x=${scaled.text.tagline.x}, y=${scaled.text.tagline.y}, fontSize=${scaled.text.tagline.fontSize}`
  );

  return scaled;
}

function generateCSSCode(proportions) {
  const css = `
/* Logo Section */
.logo-section {
  position: absolute;
  top: 100px;
  left: 100px;
  width: ${proportions.width}px;
  height: ${proportions.height}px;
}

.bracket {
  position: absolute;
  background: linear-gradient(135deg, #1A1A1A 0%, #374151 100%);
  border-radius: 2px;
}

/* Left Bracket */
.left-bracket-vertical {
  left: ${proportions.leftBracket.vertical.x}px;
  top: ${proportions.leftBracket.vertical.y}px;
  width: ${proportions.leftBracket.vertical.width}px;
  height: ${proportions.leftBracket.vertical.height}px;
}

.left-bracket-top {
  left: ${proportions.leftBracket.topHorizontal.x}px;
  top: ${proportions.leftBracket.topHorizontal.y}px;
  width: ${proportions.leftBracket.topHorizontal.width}px;
  height: ${proportions.leftBracket.topHorizontal.height}px;
}

.left-bracket-bottom {
  left: ${proportions.leftBracket.bottomHorizontal.x}px;
  top: ${proportions.leftBracket.bottomHorizontal.y}px;
  width: ${proportions.leftBracket.bottomHorizontal.width}px;
  height: ${proportions.leftBracket.bottomHorizontal.height}px;
}

/* Right Bracket */
.right-bracket-vertical {
  left: ${proportions.rightBracket.vertical.x}px;
  top: ${proportions.rightBracket.vertical.y}px;
  width: ${proportions.rightBracket.vertical.width}px;
  height: ${proportions.rightBracket.vertical.height}px;
}

.right-bracket-top {
  left: ${proportions.rightBracket.topHorizontal.x}px;
  top: ${proportions.rightBracket.topHorizontal.y}px;
  width: ${proportions.rightBracket.topHorizontal.width}px;
  height: ${proportions.rightBracket.topHorizontal.height}px;
}

.right-bracket-bottom {
  left: ${proportions.rightBracket.bottomHorizontal.x}px;
  top: ${proportions.rightBracket.bottomHorizontal.y}px;
  width: ${proportions.rightBracket.bottomHorizontal.width}px;
  height: ${proportions.rightBracket.bottomHorizontal.height}px;
}

.company-name {
  position: absolute;
  left: ${proportions.text.orson.x}px;
  top: 50%;
  transform: translateY(-50%);
  text-align: center;
}

.company-name .orson {
  font-size: ${proportions.text.orson.fontSize}px;
  font-weight: ${proportions.text.orson.fontWeight};
  color: #1A1A1A;
  margin: 0;
}

.company-name .vision {
  font-size: ${proportions.text.vision.fontSize}px;
  font-weight: ${proportions.text.vision.fontWeight};
  color: #78716C;
  margin: 0;
}

.tagline {
  position: absolute;
  left: ${proportions.text.tagline.x}px;
  top: ${proportions.text.tagline.y}px;
  transform: translateX(-50%);
  font-size: ${proportions.text.tagline.fontSize}px;
  font-weight: ${proportions.text.tagline.fontWeight};
  color: #78716C;
}
`;

  console.log(`\n=== Generated CSS ===`);
  console.log(css);

  return css;
}

// Example usage
console.log('Orson Vision Logo Proportion Calculator');
console.log('======================================');

// Test with different sizes
const sizes = [
  { width: 1200, height: 628, name: 'Google Ads' },
  { width: 1200, height: 628, name: 'Facebook Feed' },
  { width: 1200, height: 627, name: 'LinkedIn Sponsored' },
  { width: 350, height: 200, name: 'Business Card' },
  { width: 400, height: 400, name: 'LinkedIn Profile' },
];

sizes.forEach(size => {
  console.log(`\n${'='.repeat(50)}`);
  console.log(`${size.name}: ${size.width}x${size.height}px`);
  console.log(`${'='.repeat(50)}`);

  const proportions = calculateProportions(size.width, size.height);
  generateCSSCode(proportions);
});

module.exports = { calculateProportions, generateCSSCode };

File: scripts/calculate-logo-proportions.py
#!/usr/bin/env python3
"""
Logo Proportion Calculator for Orson Vision Digital Vérité Brand
Calculates exact bracket positions and text positioning for any target size
based on the original 600x200 logo proportions.
"""

import math

def calculate_logo_proportions(target_width, target_height):
    """
    Calculate exact proportions for any target size based on original 600x200 logo.

    Original logo measurements:
    - Width: 600px, Height: 200px
    - Left bracket: x=50, y=50, width=8, height=100
    - Right bracket: x=542, y=50, width=8, height=100
    - Text: centered between brackets
    """

    # Original logo dimensions
    original_width = 600
    original_height = 200

    # Original bracket measurements
    left_bracket = {
        'x': 50,
        'y': 50,
        'width': 8,
        'height': 100
    }

    right_bracket = {
        'x': 542,
        'y': 50,
        'width': 8,
        'height': 100
    }

    # Calculate scale factors
    scale_x = target_width / original_width
    scale_y = target_height / original_height

    # Scale brackets
    scaled_left_bracket = {
        'x': left_bracket['x'] * scale_x,
        'y': left_bracket['y'] * scale_y,
        'width': left_bracket['width'] * scale_x,
        'height': left_bracket['height'] * scale_y
    }

    scaled_right_bracket = {
        'x': right_bracket['x'] * scale_x,
        'y': right_bracket['y'] * scale_y,
        'width': right_bracket['width'] * scale_x,
        'height': right_bracket['height'] * scale_y
    }

    # Calculate text position (centered between brackets)
    text_x = (scaled_left_bracket['x'] + scaled_left_bracket['width'] +
              scaled_right_bracket['x']) / 2
    text_y = target_height / 2

    return {
        'container': {
            'width': target_width,
            'height': target_height
        },
        'left_bracket': scaled_left_bracket,
        'right_bracket': scaled_right_bracket,
        'text': {
            'x': text_x,
            'y': text_y
        },
        'scale_factors': {
            'x': scale_x,
            'y': scale_y
        }
    }

def generate_css(proportions, asset_name):
    """Generate CSS for the calculated proportions."""

    css = f"""/* {asset_name} - {proportions['container']['width']}x{proportions['container']['height']} */
.logo-container {{
    position: relative;
    width: {proportions['container']['width']}px;
    height: {proportions['container']['height']}px;
    background: #000;
    display: flex;
    align-items: center;
    justify-content: center;
}}

.bracket {{
    position: absolute;
    background: #fff;
}}

.left-bracket {{
    left: {proportions['left_bracket']['x']}px;
    top: {proportions['left_bracket']['y']}px;
    width: {proportions['left_bracket']['width']}px;
    height: {proportions['left_bracket']['height']}px;
}}

.right-bracket {{
    left: {proportions['right_bracket']['x']}px;
    top: {proportions['right_bracket']['y']}px;
    width: {proportions['right_bracket']['width']}px;
    height: {proportions['right_bracket']['height']}px;
}}

.logo-text {{
    position: absolute;
    left: {proportions['text']['x']}px;
    top: {proportions['text']['y']}px;
    transform: translate(-50%, -50%);
    color: #fff;
    font-family: 'Arial', sans-serif;
    font-weight: bold;
    font-size: {max(12, proportions['container']['width'] * 0.04)}px;
    text-align: center;
    white-space: nowrap;
}}"""

    return css

def generate_html(proportions, asset_name):
    """Generate complete HTML for the asset."""

    css = generate_css(proportions, asset_name)

    html = f"""<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>{asset_name}</title>
    <style>
{css}
    </style>
</head>
<body>
    <div class="logo-container">
        <div class="bracket left-bracket"></div>
        <div class="bracket right-bracket"></div>
        <div class="logo-text">ORSON VISION</div>
    </div>
</body>
</html>"""

    return html

def main():
    """Main function to calculate proportions for different asset sizes."""

    # Define target asset sizes
    assets = {
        'Google Ads Responsive': (1200, 628),
        'Facebook Feed': (1200, 628),
        'LinkedIn Sponsored Content': (1200, 627),
        'Business Card': (350, 200),
        'LinkedIn Profile': (1584, 396),
        'Email Header': (600, 150),
        'Favicon': (32, 32),
        'Loading Animation': (200, 200),
        '404 Error Page': (400, 300)
    }

    print("Orson Vision Digital Vérité Logo Proportion Calculator")
    print("=" * 60)
    print()

    for asset_name, (width, height) in assets.items():
        print(f"📐 {asset_name} ({width}x{height})")
        print("-" * 40)

        proportions = calculate_logo_proportions(width, height)

        print(f"Scale factors: X={proportions['scale_factors']['x']:.3f}, Y={proportions['scale_factors']['y']:.3f}")
        print(f"Left bracket: x={proportions['left_bracket']['x']:.1f}, y={proportions['left_bracket']['y']:.1f}, w={proportions['left_bracket']['width']:.1f}, h={proportions['left_bracket']['height']:.1f}")
        print(f"Right bracket: x={proportions['right_bracket']['x']:.1f}, y={proportions['right_bracket']['y']:.1f}, w={proportions['right_bracket']['width']:.1f}, h={proportions['right_bracket']['height']:.1f}")
        print(f"Text position: x={proportions['text']['x']:.1f}, y={proportions['text']['y']:.1f}")
        print()

        # Generate HTML file
        html_content = generate_html(proportions, asset_name)
        filename = f"orson-vision-{asset_name.lower().replace(' ', '-')}-{width}x{height}.html"

        with open(filename, 'w', encoding='utf-8') as f:
            f.write(html_content)

        print(f"✅ Generated: {filename}")
        print()

if __name__ == "__main__":
    main()
